### ISSUE TITLE
Volume video

### ISSUE THREAD

**Issue by EB-kim** [ml-struct-bio/cryodrgn#63]
Hi,
 I am trying to create a  movie file using the. volume 0.mrc to 9.mrc just like the video shown in the tutorial, but I can't find in the tutorial how to do it or in the paper. Can you tell me how you created the video?

**Comment by Guillawme** [comment]
Hi,

I was just doing this recently, so here are a few [ChimeraX](https://www.cgl.ucsf.edu/chimerax/) commands to get you started. You would of course need to also adjust the orientation of your map to show a view that makes sense, and I recommend doing this with the [`turn`](https://www.cgl.ucsf.edu/chimerax/docs/user/commands/turn.html) command so it is in your ChimeraX command file and therefore reproducible (it's near impossible to find the exact same orientation again by hand, if you don't automate it from the very beginning; and when you make figures it's nice to have reproducible orientations, or else you have to remake every single panel whenever the orientation changes slightly because you did not have it scripted).

```
# Open your maps as a volume series
open vol_???.mrc vseries true

# Global display options
set bgcolor white
lighting soft
graphics silhouettes true
# Use same color as in the cryoDRGN paper
volume #1 color #6096fe

# Adjust map contour level and optionally remove speckles
# Adjust these numbers to whatever makes sense in your specific case
volume #1 level 0.01
surface dust #1 size 10

# Generate a movie going through the volume series

# The 'graphics rate maxframerate 30' command causes the resulting movie to be at 30 frames per second
# instead of whatever the default is in ChimeraX (I think it's 60); I chose 30 fps
# because it is the default export frame rate in Openshot, the movie editing software I subsequently 
# used to assemble several movie files into a single movie

# To adjust the number of steps through the volume series per second, you need to
# play with the 'pauseframes 15' and 'wait 900' numbers:
# I had 59 volumes and wanted to go through them at 2 volumes per second
# so 'pauseframes 15' keeps each step in the volume series displayed for 15 movie frames (so 0.5 s at 30 fps)
# before changing to the next volume
# and 'wait 900' makes the entire movie 900 frames (so 30 s at 30 fps)

view #1
graphics rate maxframerate 30
#Forward direction movie
vseries play #1 jumpto 0
movie record
vseries play #1 normalize true pauseframes 15; wait 900
movie stop
movie encode forward.mp4 framerate 30
# Reverse direction movie
vseries play #1 jumpto 9
movie record
vseries play #1 normalize true direction backward pauseframes 15; wait 900
movie stop
movie encode backward.mp4 framerate 30
```

When the corresponding paper is published (will probably take some more time...), I am planning to add my complete ChimeraX command files to [this repository](https://github.com/Guillawme/chimerax-viz), so keep an eye on it if you're interested.

I hope this helps, good luck.

**Comment by Guillawme** [comment]
@zhonge : since questions like this one that are not bug reports start coming in, do you think we should use the "discussions" feature of GitHub? (if yes, you would need to activate it in the repository settings). Seems like it could be helpful to keep bug reports easy to browse while also allowing usage questions (the Topaz repository uses it, if you want to see what it's like).

**Comment by EB-kim** [comment]
I think it would be helpful to use discussion feature. I have chimera rather than chimeraX, I don't know why but I am having little trouble with it. Can I run this on Chimera or only on ChimeraX?

**Comment by zhonge** [comment]
Thanks for your contributions and that sounds like a great idea! I will set it up. 

A couple of additional details on how I typically generate the movies (which I can repeat in the discussion once that is set up):
```
# use the same lighting settings as @Guillawme mentioned above

volume all color cornflowerblue # a named color 
volume all level XX # common isosurface level

movie record
mseries #1-10
movie encode FILENAME.mp4 framerate 10 
```

I use framerate 10 for continuous trajectories and framerate 3 for discrete samples. I am definitely not a chimerax movie making expert so looking forward to seeing other suggestions.

**Comment by Guillawme** [comment]
Ha, thank you for the color name! I picked it from the PDF of your paper using the color picker tool, which is why I have some hexadecimal number.

I did not know about the `mseries` command. From its help page it looks very similar to `vseries`, but it seems to me that `vseries` is more appropriate for maps (volumes), because `vseries play` has this `normalize` option that keeps the volume enclosed by the displayed surface constant as you step through the volume series (by dynamically adjusting the contour level, I suppose). This is more precise than setting the same contour level for all maps, because if a part of the map becomes much weaker as you go one step into the volume series, then you're now looking at a smaller volume than you had for the previous step.

Fiddling with the frame rate was the most difficult part, because the explanations in the ChimeraX documentation are a bit complicated. It doesn't matter for a movie only showing a map, but if you want to have in the same movie an animation with dots on a PCA or UMAP plot, then obviously both must have the same frame rate (and same "step" rate, so to speak, through the series).
